Backend/BlockChangesService.cs

```diff
@@ -12,6 +13,7 @@ namespace FRITeam.Swapify.Backend
     public class BlockChangesService : IBlockChangesService
     {
         private readonly IMongoCollection<BlockChangeRequest> _blockChangesCollection;
+        private static ConcurrentDictionary<BlockChangeRequest, object> _lockObjects = new ConcurrentDictionary<BlockChangeRequest, object>();
 
```

Comment:
BlockChangeRequest doesn't implement GetHasCode and equals, so the key will be different for 2 instances of the same object (key is based on reference)...
Read: https://blog.markvincze.com/back-to-basics-dictionary-part-3-built-in-gethashcode/